[Merged by Bors] - feat(linear_algebra/basic): add comap span lemmas

ocfnash opened the pull request:
We already had `map_span` but nothing for `comap`.

ocfnash committed:
feat(linear_algebra/basic): add comap span lemmas

We already had `map_span` but nothing for `comap`.

jcommelin commented on src/linear_algebra/basic.lean:
```diff
@@ -787,6 +787,11 @@ lemma map_span (f : M →ₗ[R] M₂) (s : set M) :
 map_le_iff_le_comap.2 $ span_le.2 $ λ x hx, subset_span ⟨x, hx, rfl⟩
 
+/- See also `comap_span_eq` below. -/
+lemma comap_span_le (f : M →ₗ[R] M₂) (s : set M₂) :
```
```suggestion
lemma span_preimage_le (f : M →ₗ[R] M₂) (s : set M₂) :
```

ocfnash committed:
CR suggestion: rename `comap_span_le` --> `span_preimage_le`

Likewise renaming `comap_span_eq` --> `span_preimage_eq` and
moving `span_preimage_eq` from `linear_map` to `submodule` namespace
for consistency.
